### DISCUSSION TITLE
Giving entities human readable names and find entity by name at runtime

### DISCUSSION THREAD

**Discussion by KabelitzJ** [skypjack/entt#797] [Q&A]
When creating an editor for a scene view it is helpful to give entities human readable and unique names.
How would I implement such a feature using `entt`? Or is that something that has nothing to do with the `entt::registry` but rather a e.g. `std::unordered_map<std::string, entt::entity>` that maps entities to names.
Though this has the disadvantage of having to manage an entity in two seperate places.

**Comment by Innokentiy-Alaytsev** [comment]
You can assign a name as a component and implement uniqueness testing upon every component assignment.

**Reply by KabelitzJ** [reply]
Who would I create a component out of a runtime string?

**Reply by Innokentiy-Alaytsev** [reply]
```c++
struct EntityName {
    std::string value;
};


reg.emplace< EntityName > (my_entity, "Human-readable Name");
```

Or in any other way that allows storing a runtime string.

**Reply by KabelitzJ** [reply]
But then, to find a entity by its name I would have to query all `EntityName`-components. When I go so far as giving all entities a name I would have to loop over all existing entities.

**Reply by KabelitzJ** [reply]
If there is no other way, thats also fine. I just thought someone smarter that me had run into the same problem and figured out a solution.
Because entities with names seems like a common use case.

**Comment by skypjack** [comment]
It's not that common actually but there is nothing wrong. Though, a registry is an entity-to-component mapping while what you want is a name-to-entity mapping, that is exactly the opposite. 🙂 
Also, there are some corner cases to address, i.e. what happens when an entity gets destroyed? Should it retain its name because the latter doesn't depend on the entity version or should it release the name?
I'm curious to know what your thoughts about it are. Giving entities a name sounds pretty easy otherwise.
